share/availability/full/reconstruction_test.go

```diff
@@ -176,100 +255,141 @@ func TestShareAvailable_DisconnectedFullNodes(t *testing.T) {
 	// └─┴─┤   ├─┴─┘
 	//    F└───┘F
 	//
-
-	// NOTE: Numbers are taken from the original 'Fraud and Data Availability Proofs' paper
-	light.DefaultSampleAmount = 20 // s
-	const (
-		origSquareSize = 16 // k
-		lightNodes     = 32 // c - total number of nodes on two subnetworks
-	)
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
-	defer cancel()
-
-	net := availability_test.NewTestDAGNet(ctx, t)
-	source, root := RandNode(net, origSquareSize)
-
-	// create light nodes and start sampling for them immediately
-	lights1, lights2 := make(
-		[]*availability_test.TestNode, lightNodes/2),
-		make([]*availability_test.TestNode, lightNodes/2)
-
-	var wg sync.WaitGroup
-	wg.Add(lightNodes)
-	for i := 0; i < len(lights1); i++ {
-		lights1[i] = light.Node(net)
-		go func(i int) {
-			defer wg.Done()
-			err := lights1[i].SharesAvailable(ctx, root)
-			if err != nil {
-				t.Log("light1 errors:", err)
+	tc := []struct {
+		name            string
+		origSquareSize  int  // k
+		lightNodes      int  // c
+		sampleAmount    uint // s
+		recoverability  Recoverability
+		expectedFailure bool
+	}{
+		{
+			name:            "fully recoverable",
+			origSquareSize:  16,
+			lightNodes:      32,
```

Comment:
In TestShareAvailable_ConnectedFullNodes this value is set to 24. Whats is the math behind it being different here?

Reply:
This is an artifact from the old math, which was 16*2, but this is actually incorrect as it is more than necessary. Thanks for the catch